xrally_kubernetes/platforms/existing.py

```diff
@@ -12,30 +12,55 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
-from rally.common import cfg
-from rally.common import logging
+import json
+import traceback
+
+from kubernetes import config
+from kubernetes.client.apis import core_v1_api
+from kubernetes.client.apis import version_api
 from rally.env import platform
 
 
-LOG = logging.getLogger(__name__)
+@platform.configure(name="existing", platform="kubernetes")
+class KubernetesPlatform(platform.Platform):
+    """Default plugin for Kubernetes."""
 
-CONF = cfg.CONF
+    CONFIG_SCHEMA = {
+        "type": "object",
+        "properties": {
+            "config_file": {
+                "type": "string"
+            }
+        },
+        "additionalProperties": False
+    }
 
+    def __init__(self, spec, **kwargs):
+        super(KubernetesPlatform, self).__init__(spec, **kwargs)
+        self._api = None
 
-@platform.configure(name="existing", platform="kubernetes")
-class Kubernetes(platform.Platform):
-    """Default plugin for Kubernetes platform
-
-    It may be used to test any existing Kubernetes API compatible cluster.
-    """
     def create(self):
-        # TBD
-        return {}, {}
+        """Pass kubeconfig to internal Configuration."""
+        return {"config_file": self.spec.get("config_file")}, {}
 
     def destroy(self):
-        # NOTE(boris-42): No action need to be performed.
+        # NOTE(prazumovsky): No action need to be performed.
         pass
 
+    def check_health(self):
+        """Check whatever platform is alive."""
+        try:
+            api = config.new_client_from_config(self.spec.get("config_file"))
+            core_v1_api.CoreV1Api(api_client=api).list_namespace()
```

Comment:
actually, let's call just ``self.info()``. it already calls to k8s api and contains k8s client initialization